packages/rocketchat-livechat/client/views/app/tabbar/visitorInfo.js

```diff
@@ -243,18 +250,17 @@ Template.visitorInfo.onCreated(function() {
 
 	if (currentData && currentData.rid) {
 		this.autorun(() => {
-			const room = ChatRoom.findOne(currentData.rid);
-			if (room && room.v && room.v._id) {
-				this.visitorId.set(room.v._id);
-			} else {
-				this.visitorId.set();
-			}
+			const room = Rooms.findOne(currentData.rid);
+			this.visitorId.set(room && room.v && room.v._id);
+			this.departmentId.set(room && room.departmentId);
 		});
 
 		this.subscribe('livechat:visitorInfo', { rid: currentData.rid });
+		this.subscribe('livechat:departments');
 	}
 
 	this.autorun(() => {
 		this.user.set(LivechatVisitor.findOne({ _id: this.visitorId.get() }));
+		this.department.set(LivechatDepartment.findOne({ _id: this.departmentId.get() }));
```

Comment:
if you'll use the department only in one helper, it is better to just return the find on the helper itself. having a reactive var like this is better if you need to use the value in multiples places, so you don't need to do multiple finds which saves in performance.

Reply:
Fixed.